src/monattack.cpp

```diff
@@ -3750,12 +3750,28 @@ bool mattack::longswipe(monster *z)
     if( target == nullptr ) {
         return false;
     }
+    if( rl_dist( z->pos(), target->pos() ) > 3 || !z->sees( *target ) ) {
+        return false; //out of range
+    }
+    //Is there something impassable blocking the claw?
+    std::vector<tripoint> line = g->m.find_clear_path( z->pos(), target->pos() );
+    for( auto &i : line ){
+        ter_t terrain = g->m.ter_at( i );
+        if( terrain.movecost == 0 ){
```

Comment:
There is `map::impassable` for that. Checking only the terrain will skip furniture and vehicle boards.